samples/demo-app/app/src/main/java/cash/z/ecc/android/sdk/demoapp/demos/getbalance/GetBalanceFragment.kt

```diff
@@ -0,0 +1,87 @@
+package cash.z.ecc.android.sdk.demoapp.demos.getbalance
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import androidx.lifecycle.lifecycleScope
+import cash.z.ecc.android.bip39.Mnemonics
+import cash.z.ecc.android.bip39.toSeed
+import cash.z.ecc.android.sdk.Initializer
+import cash.z.ecc.android.sdk.Synchronizer
+import cash.z.ecc.android.sdk.block.CompactBlockProcessor
+import cash.z.ecc.android.sdk.demoapp.App
+import cash.z.ecc.android.sdk.demoapp.BaseDemoFragment
+import cash.z.ecc.android.sdk.demoapp.databinding.FragmentGetBalanceBinding
+import cash.z.ecc.android.sdk.ext.collectWith
+import cash.z.ecc.android.sdk.ext.convertZatoshiToZecString
+import cash.z.ecc.android.sdk.tool.DerivationTool
+
+class GetBalanceFragment : BaseDemoFragment<FragmentGetBalanceBinding>() {
+
+    private lateinit var synchronize: Synchronizer
+
+    override fun inflateBinding(layoutInflater: LayoutInflater): FragmentGetBalanceBinding =
+        FragmentGetBalanceBinding.inflate(layoutInflater)
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        setup()
+    }
+
+    private fun setup() {
+        // defaults to the value of `DemoConfig.seedWords` but can also be set by the user
+        val seedPhrase = sharedViewModel.seedPhrase.value
+
+        // Use a BIP-39 library to convert a seed phrase into a byte array. Most wallets already
+        // have the seed stored
+        val seed = Mnemonics.MnemonicCode(seedPhrase).toSeed()
+
+        // converting seed into viewingKey
+        val viewingKey = DerivationTool.deriveViewingKeys(seed).first()
+
+        // using the ViewingKey to initialize
+        App.instance.defaultConfig.let { config ->
+            Initializer(App.instance) {
+                it.importWallet(viewingKey, config.birthdayHeight)
+                it.server(config.host, config.port)
+            }.let { initializer ->
+                synchronize = Synchronizer(initializer)
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        // the lifecycleScope is used to dispose of the synchronize when the fragment dies
+        synchronize.start(lifecycleScope)
+        monitorChanges()
+    }
+
+    private fun monitorChanges() {
+        synchronize.status.collectWith(lifecycleScope, ::onStatus)
+        synchronize.balances.collectWith(lifecycleScope, ::onBalance)
+    }
+
+    private var isSyncing = true
+
+    private fun onBalance(balance: CompactBlockProcessor.WalletBalance) {
+        this.balance = balance
+        if (!isSyncing) {
```

Comment:
this works and is totally fine! However, I only mention this just to add to your familiarity with the API:

The value of balance shouldn't receive an event until a balance is calculated. In theory, the first balance you get from this flow should be -1 (because the channel is [initialized with an empty WalletBalance](https://github.com/zcash/zcash-android-wallet-sdk/blob/master/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt#L51) object which [defaults to -1](https://github.com/zcash/zcash-android-wallet-sdk/blob/master/src/main/java/cash/z/ecc/android/sdk/block/CompactBlockProcessor.kt#L640-L643)) and then the very next balance shouldn't come until the wallet is synced (because [refreshBalances](https://github.com/zcash/zcash-android-wallet-sdk/blob/master/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt#L272-L275) isn't called until the [full scan has completed](https://github.com/zcash/zcash-android-wallet-sdk/blob/master/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt#L360-L385))

Now, whether it behaves that way in reality... I don't recall, without testing :laughing: 
but it should
and if it doesn't we should fix it :grin:

Reply:
we could maybe add an extension function to the walletbalance object that could help you write something like this:

```
// extension property `none` returns true when total and available equal -1
if (balance.none) {
    // show "calculating balance" message
} else {
    // show balance
} 
```

Reply:
`isSet` might actually be a better name for that property